src/main/kotlin/com/ampnet/auditornode/service/impl/KethabiAuditRegistryContractTransactionService.kt

```diff
@@ -0,0 +1,55 @@
+package com.ampnet.auditornode.service.impl
+
+import com.ampnet.auditornode.configuration.properties.AuditorProperties
+import com.ampnet.auditornode.contract.AuditRegistryContractTransactionGenerator
+import com.ampnet.auditornode.persistence.model.AssetContractAddress
+import com.ampnet.auditornode.persistence.model.UnsignedTransaction
+import com.ampnet.auditornode.script.api.model.AbortedAudit
+import com.ampnet.auditornode.script.api.model.AuditResult
+import com.ampnet.auditornode.script.api.model.FailedAudit
+import com.ampnet.auditornode.script.api.model.SuccessfulAudit
+import com.ampnet.auditornode.service.AuditRegistryContractTransactionService
+import mu.KotlinLogging
+import org.kethereum.model.Address
+import org.komputing.khex.extensions.toHexString
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+class KethabiAuditRegistryContractTransactionService @Inject constructor(
+    auditorProperties: AuditorProperties
+) : AuditRegistryContractTransactionService {
+
+    private val contractAddress = Address(auditorProperties.auditRegistryContractAddress)
+    private val contractTransactionGenerator = AuditRegistryContractTransactionGenerator(contractAddress)
+
+    override fun castAuditVoteForAsset(
```

Comment:
Maybe some name with generateTx, `generateTxForAssetAuditVote`? Is cast some specific term in this case?

Reply:
`castVote` is the name of the contract method that Filip used. What do you think about naming it `generateTxForCastAuditVote`?

Reply:
Sounds good, provides much more info about function to me.